integration-test/test/handler-commands.bats

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env bats
+
+load test_helper
+
+setup(){
+    build_docker_image
+}
+
+teardown(){
+    rm -rf "$certs_dir"
+}
+
+@test "handler command: install - creates the data dir" {
+    run in_container fake-waagent install
+    echo "$output"
+    [ "$status" -eq 0 ]
+    [[ "$output" = *event=installed* ]]
+
+    diff="$(container_diff)"
+    echo "$diff"
+    [[ "$diff" = *"A /var/lib/azure/custom-script"* ]]
+}
+
+@test "handler command: enable - can process empty settings, but fails" {
+    mk_container sh -c "fake-waagent install && fake-waagent enable"
+    push_settings '' ''
+
+    run start_container
+    echo "$output"
+    [ "$status" -eq 1 ]
+    [[ "$output" == *"invalid configuration: 'commandToExecute' is not specified"* ]]
```

Comment:
How valuable is it to verify the error message in detail at this stage.  I feel like you have numerous tests to catch this already, so why do it again.  Is error code sufficient?